MediaAdminModelBundle/Migrations/MongoDB/Version20160411173754.php

```diff
@@ -0,0 +1,779 @@
+                    && item.attributes.' . $attributeName . '.type == "' . self::TINYMCE_TYPE . '"
+                ) {
+                    tinymceValue = item.attributes.' . $attributeName . '.value;';
+
+            foreach (self::ALTERNATIVES_RENAMING as $oldAlternativeName => $newAlternativeName) {
+                $query .= '
+                    tinymceValue = tinymceValue.replace("[media=' . $oldAlternativeName . ']", "[media=' . $newAlternativeName . ']");';
+            }
+
+        $query .= '
+                    item.attributes.' . $attributeName . '.value = tinymceValue;
+                    item.attributes.' . $attributeName . '.stringValue = tinymceValue;
+                }
+
+                db.content.update({_id: item._id}, item);
+            });';
+        }
+
+        return $query;
+    }
+
+    /**
+     * Get the request to update the nodes by redefining the media id storage
+     *
+     * @param array blockType
+     * @param array attributes
+     *
+     * @return string
+     */
+    protected function getUpdateNodesRequest($blockType, $attributes)
+    {
+        $filters = '
+            {
+                "blocks": {
+                    $elemMatch: {"component": "' . $blockType . '"}
+                }
+            }';
+
+        $query = '
+            db.node.find(' . $filters . ').forEach(function(item) {
+
+                function formatMediaId(mediaId) {
+                    if (mediaId.indexOf("' . self::MEDIA_PREFIX . '") == 0) {
+                        mediaId = mediaId.replace("' . self::MEDIA_PREFIX . '", "");
+                    }
+                    return {
+                        "id": mediaId,
+                        "format": ""
+                    };
+                }
+
+                blocks = item.blocks;
+
+                for (var i =  0; i < blocks.length; i++) {
+                    if (blocks[i]["component"] == "' . $blockType . '") {';
+
+        foreach ($attributes as $attributeName) {
+            $query .= '
+                        mediaAttribute = blocks[i]["attributes"]["' . $attributeName . '"];
+                        if (typeof mediaAttribute != undefined) {
+                            if (typeof mediaAttribute == "string") {
+                                mediaAttribute = formatMediaId(mediaAttribute);
+                            } else if (mediaAttribute.constructor === Array) {
+                                for (var j = 0; j < mediaAttribute.length; j++) {
+                                    if (typeof mediaAttribute[j] == "string") {
+                                        mediaAttribute[j] = formatMediaId(mediaAttribute[j]);
+                                    }
+                                }
+                            }
+                        }';
+        }
+
+        $query .= '
+                        blocks[i]["attributes"]["' . $attributeName . '"] = mediaAttribute;
```

Comment:
if for any reason the variable attributes is empty, the attributeName doesn't exist